Argo/Types/Decoded/Alternative.swift

```diff
@@ -13,8 +13,15 @@
 infix operator <|> { associativity left precedence 140 }
 
 public func <|> <T>(lhs: Decoded<T>, rhs: Decoded<T>) -> Decoded<T> {
-  if case .Success = lhs {
-    return lhs
+  return lhs.alternative(rhs)
+}
+
+public extension Decoded {
+  func alternative<T>(other: Decoded<T>) -> Decoded<T> {
+    if case .Success = self {
```

Comment:
Ah, I see. That's because you're actually declaring a new type `T`. Subtle bug. Make this change on line 20:

```diff
-  func alternative<T>(other: Decoded<T>) -> Decoded<T> {
+  func alternative(other: Decoded<T>) -> Decoded<T> {
```

and it should all work